Math-Interactions-Overview.md

```diff
@@ -0,0 +1,49 @@
+There are currently three interactions that can be used for math-related questions in lessons:
+- AlgebraicExpressionInput
+- NumericExpressionInput
+- MathEquationInput
+
+## Code structure
+### Frontend
+The interaction-specific files are in `extensions/interactions`. Each interaction has its own directory. The general structure of this directory would be as follows:
+
+![interactions](https://user-images.githubusercontent.com/35144226/228078766-a0fb8b8a-a03b-4df1-add5-59725d41cdeb.PNG)
+
+The `core/templates/services` contains the following services related to the math interactions:
+- `guppy-initialization-service.ts`: Service for initializing a guppy instance. On the creator's dashboard, there could be multiple guppy instances present at any given point. This service manages all of these instances and helps in initializing a new instance and finding the currently active instance.
+- `guppy-configuration-service.ts`: Service for handling the global config that will be applied to all guppy instances. 
+- `math-interaction-service.ts`: The core service for handling most of the math interaction-related functionalities on the frontend. This service handles:
+  - All frontend validations using nerdamer.
+  - Modifying the user input from guppy and transforming it such that it is compatible with nerdamer and our backend parser schema.
+  - Generating error messages for various invalid scenarios.
+
+For the custom on-screen keyboard, the functionality is handled by the component present in `core/templates/components/on-screen-keyboard`. This does not extend the custom OSK provided by guppy since customization on that is quite limited. This contains normal buttons that call the guppy API functions to insert characters onto the currently active guppy instance.
```

Comment:
This contains normal --> The custom on-screen keyboard contains normal

Reply:
Done.